packages/xod-project/src/builtInPatches.js

```diff
@@ -44,14 +49,21 @@ const getTerminalPins = R.curry((direction, type) => {
     0, // order
     label,
     '', // description
-    true, // bindable? terminal's pins are always bindable
-    DEFAULT_VALUE_OF_TYPE[type]
+    // bindable? terminal's pins are always bindable for built-in types
+    isBuiltInType(type),
+    R.propOr('', type, DEFAULT_VALUE_OF_TYPE)
   );
 
   return R.objOf(pinKey, pin);
 });
 
+export const getCustomTypeTerminalPins = R.converge(getTerminalPins, [
+  getTerminalDirection,
+  getTerminalDataType,
+]);
+
 const BUILT_IN_MARKERS = [
+  OUTPUT_SELF_PATH,
```

Comment:
:thinking: output-self is not a marker. `PINS_FOR_BUILT_IN_MARKERS` and `PINS_OF_PATCH_NODES` will work incorrectly

Reply:
If it's a terminal, than it's a *very* special one.
- We do not have `self` type
- Nothing can be linked/bound to `output-self`
- It does not have an `input` counterpart

I think it kind of *is* a marker 🤔

Reply:
Good points. Initially, I thought about a future where we can derive new types from tuples by linking them to an `output-self`. But, yes, we are not even close to it. So, OK, let it be a marker now.